### PR TITLE
Averaged fan adjoint new

### PR DESCRIPTION
**Pull request by roystgnr**
This reapplies the changes to master, and adds a new post-bugfix gold standard, and so supercedes #173

### CODE REVIEW

File: src/physics/include/grins/averaged_fan_adjoint_stab.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    affected by a region in which airfoils are moving.
+   */
+  template<class Viscosity>
+  class AveragedFanAdjointStabilization : public AveragedFanBase<Viscosity>
+  {
+  public:
+
+    AveragedFanAdjointStabilization( const std::string& physics_name, const GetPot& input );
+
+    ~AveragedFanAdjointStabilization();
```
Comment (pbauman) [inline src/physics/include/grins/averaged_fan_adjoint_stab.h]: Probably want to make this virtual.

File: src/physics/include/grins/averaged_fan_base.h [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+{
+
+  template<class Viscosity>
+  class AveragedFanBase : public IncompressibleNavierStokesBase<Viscosity>
+  {
+  public:
+
+    AveragedFanBase( const std::string& physics_name, const GetPot& input );
+
+    ~AveragedFanBase();
```
Comment (pbauman) [inline src/physics/include/grins/averaged_fan_base.h]: Ditto.

**Comment by pbauman** [comment]
OK, otherwise (I just scanned the diffs though).

**Comment by roystgnr** [comment]
Those are actually already virtual - Physics::~Physics is declared virtual and everything inherits that.  (IMHO it was a bug until C++11 added the "final" keyword that you *couldn't* undo a base class virtual.   Might be good to take a sweep through the code and make it explicit, though.

**Comment by roystgnr** [comment]
By "bug" I mean "flaw", of course.  Also, I'd like to postpone sweeps through the code (there's also some messed up spacing in some license boilerplate) until after I've got these extra physics classes merged; it's tricky enough updating everything to match the FEMSystem change.  Might even postpone until after SA is added to avoid messing with Vikram too badly.

**Comment by pbauman** [comment]
Agreed on all counts.

**Comment by pbauman** [comment]
Should've probably added merge when you're ready.

**Pull request merged**